presto-main/src/test/java/com/facebook/presto/sql/planner/TestPredicatePushdown.java

```diff
@@ -267,4 +268,51 @@ public void testPredicatePushDownThroughMarkDistinct()
                                 project(assignUniqueId("unique", filter("A = 1", values("A")))),
                                 project(filter("1 = B", values("B"))))));
     }
+
+    @Test
+    public void testPredicatePushDownOverProjection()
+    {
+        // Non-singletons should not be pushed down
+        assertPlan(
+                "with t as (select orderkey * 2 x from orders) select * from t where x + x > 1",
```

Comment:
I think placing the `with` and the `select` in separate lines will improve readability:

```java
assertPlan(
    "with t as (select orderkey * 2 x from orders) " + 
    "select * from t where x + x > 1",
    anyTree(...));
```